packages/core/utils/src/prettyDiagnostic.js

```diff
@@ -10,10 +10,18 @@ import nullthrows from 'nullthrows';
 // $FlowFixMe
 import terminalLink from 'terminal-link';
 
+export type FormattedCodeFrame = {|
+  location: string,
+  code: string,
+|};
+
 export type AnsiDiagnosticResult = {|
   message: string,
   stack: string,
+  /** A formatted string containing all code frames, including their file locations. */
   codeframe: string,
+  /** A list of code frames with highlighted code and file locations separately. */
+  frames: Array<FormattedCodeFrame>,
```

Comment:
A bit unfortunate, but couldn't think of a better way...

Reply:
You could make `codeframe` a getter that processes `frames` in the default manner so there's less duplication, but beyond disgusting regex I don't think there's any other way to support rewriting paths.